BLUESPAWN-client/src/util/log/XMLSink.cpp

```diff
@@ -0,0 +1,162 @@
+			auto Cmd = XMLDoc.NewElement("hash");
+			auto Pid = XMLDoc.NewElement("pid");
+			auto Tid = XMLDoc.NewElement("tid");
+			Name->SetText(WidestringToString(ProcessDetection->wsImageName).c_str());
+			Path->SetText(WidestringToString(ProcessDetection->wsImagePath).c_str());
+			Cmd->SetText(WidestringToString(ProcessDetection->wsCmdline).c_str());
+			Pid->SetText(std::to_string(ProcessDetection->PID).c_str());
+			Tid->SetText(std::to_string(ProcessDetection->TID).c_str());
+			detect->InsertEndChild(Name);
+			detect->InsertEndChild(Path);
+			detect->InsertEndChild(Cmd);
+			detect->InsertEndChild(Pid);
+			detect->InsertEndChild(Tid);
+		} else if(detection->Type == DetectionType::Service){
+			detect->SetAttribute("type", "Service");
+			auto ServiceDetection = std::static_pointer_cast<SERVICE_DETECTION>(detection);
+			auto Name = XMLDoc.NewElement("name");
+			auto Path = XMLDoc.NewElement("path");
+			auto Dll = XMLDoc.NewElement("dll");
+			auto Pid = XMLDoc.NewElement("pid");
+			Name->SetText(WidestringToString(ServiceDetection->wsServiceName).c_str());
+			Path->SetText(WidestringToString(ServiceDetection->wsServiceExecutablePath).c_str());
+			Dll->SetText(WidestringToString(ServiceDetection->wsServiceDll).c_str());
+			Pid->SetText(std::to_string(ServiceDetection->ServicePID).c_str());
+			detect->InsertEndChild(Name);
+			detect->InsertEndChild(Path);
+			detect->InsertEndChild(Dll);
+			detect->InsertEndChild(Pid);
+		} else if(detection->Type == DetectionType::Event){
+			detect->SetAttribute("type", "Event");
+			auto EventDetection = std::static_pointer_cast<EVENT_DETECTION>(detection);
+			auto ID = XMLDoc.NewElement("id");
+			auto RecordID = XMLDoc.NewElement("recordid");
+			auto Time = XMLDoc.NewElement("time");
+			auto Channel = XMLDoc.NewElement("channel");
+			auto Raw = XMLDoc.NewElement("raw");
+			ID->SetText(std::to_string(EventDetection->eventID).c_str());
+			RecordID->SetText(std::to_string(EventDetection->eventRecordID).c_str());
+			Time->SetText(WidestringToString(EventDetection->timeCreated).c_str());
+			Channel->SetText(WidestringToString(EventDetection->channel).c_str());
+			Raw->SetText(WidestringToString(EventDetection->rawXML).c_str());
+			detect->InsertEndChild(ID);
+			detect->InsertEndChild(RecordID);
+			detect->InsertEndChild(Time);
+			detect->InsertEndChild(Channel);
+			detect->InsertEndChild(Raw);
+			for(auto key : EventDetection->params){
+				auto name = WidestringToString(key.first);
+				auto idx1 = name.find("'") + 1;
+				auto tag = XMLDoc.NewElement(name.substr(idx1, name.find_last_of("'") - idx1).c_str());
+				tag->SetText(WidestringToString(key.second).c_str());
+				detect->InsertEndChild(tag);
+			}
+		}
+		return detect;
+	}
+
+	void XMLSink::LogMessage(const LogLevel& level, const std::string& message, const std::optional<HuntInfo> info, const std::vector<std::shared_ptr<DETECTION>>& detections){
+		auto mutex = AcquireMutex(hMutex);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'mutex' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5011561)